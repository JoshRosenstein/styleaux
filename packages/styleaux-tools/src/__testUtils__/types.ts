export type Arg1<T> = T extends (arg1: infer U, ...args: any[]) => any
  ? U
  : undefined
