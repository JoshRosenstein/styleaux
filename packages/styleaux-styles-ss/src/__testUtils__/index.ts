import { flatten, toArray, mergeAllDeepRight } from "@roseys/futils";

export const toStyles = styles => mergeAllDeepRight([{}, ...flatten(toArray(styles))]);
