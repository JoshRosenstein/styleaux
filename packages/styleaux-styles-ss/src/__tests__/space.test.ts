import {space,defaultSpaceScale} from '../'
import { toStyles} from "../__testUtils__";

const MEDIA = {
  D: '(min-width: 1025px)',
  T: '(min-width: 601px) and (max-width: 1024px)',
  M: '(max-width: 600px)',
  C: '(max-width: 666px)',
}
type Imedia=typeof MEDIA

const MQ= (Object.keys(MEDIA).reduce((acc,k)=>{acc[k]='@media ' + MEDIA[k]
return acc
},{} as Imedia))

const spaceT=  {
none: 0,
xxs: 2,
xs: 4,
sm: 8,
md: 16,
lg: 32,
xl: 100,
xxl: 128,}

const theme = {
  media: MEDIA,
  space: spaceT

}

const THEME = theme;
type Options = {
  theme?: any;
};

type V = {
  values: string[];
  falseValue?: string;
  trueValue?: string;
};

const data = [
  [
    'Uses Theme',
    { m: 'xs' },
    {
      margin: '4px',
    }
  ],
  [
    'Responsive',
    { m: {all:1,T:2} },
    {
      margin: defaultSpaceScale[1]+'px', [MQ.T]:{margin:defaultSpaceScale[2]+'px'}
    }
  ],

  [
    'Responsive themekey lookup',
    { m: {all:'xs',T:'md'} },
    {
      margin: '4px', [MQ.T]:{margin:'16px'}
    }
  ],

  [
    'Shorthands overide longs',
    { mx: 'xs', ml:'lg' },
    {
      marginLeft: '32px',  marginRight: '4px',
    }
  ],

];

//@ts-ignore
test.each(data)(
  '%s',
  //@ts-ignore
  (testName: string, props: any, expected: any, theme?: any = THEME) => {
    //expect(theme).toEqual( THEME);
    expect(toStyles(space({ theme, ...props }))).toEqual(expected);
  }
);
