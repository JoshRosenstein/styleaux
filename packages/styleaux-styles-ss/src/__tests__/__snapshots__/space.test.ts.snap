// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsive 1`] = `
Object {
  "@media (min-width: 601px) and (max-width: 1024px)": Object {
    "marginBottom": "8px",
    "marginLeft": "8px",
    "marginRight": "8px",
    "marginTop": "8px",
  },
  "marginBottom": "4px",
  "marginLeft": "4px",
  "marginRight": "4px",
  "marginTop": "4px",
}
`;

exports[`Responsive themekey lookup 1`] = `
Object {
  "@media (min-width: 601px) and (max-width: 1024px)": Object {
    "marginBottom": "16px",
    "marginLeft": "16px",
    "marginRight": "16px",
    "marginTop": "16px",
  },
  "marginBottom": "4px",
  "marginLeft": "4px",
  "marginRight": "4px",
  "marginTop": "4px",
}
`;

exports[`Shorthands overide longs 1`] = `
Object {
  "marginLeft": "32px",
  "marginRight": "4px",
}
`;

exports[`Uses Theme 1`] = `
Object {
  "marginBottom": "4px",
  "marginLeft": "4px",
  "marginRight": "4px",
  "marginTop": "4px",
}
`;
